Homework19 tests

ochaklosh opened the pull request:
Method to Delete Playlist

ochaklosh committed:
Update build.gradle

Added Homework19Tests

ochaklosh committed:
Create Homework19Tests.java

deletePlaylist test

ochaklosh committed:
Update TestNG.xml

Added Homework19Tests

Berichicko commented:
Hello,

I have checked your homework 19 and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

![Thumbs-up-hand-with-crown-illustration-on-transparent-background-PNG](https://github.com/Tootienahs/A59-QA-Automation-Framework/assets/104761482/20861dce-f8e7-4435-9b92-de6436933d0b)

![Quote](https://github-readme-quotes-bay.vercel.app/quote?theme=dark&layout=socrates)


![image](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/6cb12f29-a13f-4152-b93e-d1f928c30af7)

HINT

![image](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/e21deddb-6c9a-43ed-a7c1-1669a796dc68)

https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/232ddda0-432d-42a2-8aad-d002c5ceac1a

click

![Screenshot_1](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/2d0d3a39-ed9f-4bfb-9f9d-ee9e6d9929b9)

ochaklosh commented:
Hello Alex,
I am sorry to bother you, but when I run deletePlaylist() I get a Tests Failed. Please look at the screen shot.
It can't find selector. However, for you test worked. I am confused.
I would appreciate if you could help me with this.
Thanks,
Oksana

[cid:827a6050-b990-45f1-9dec-dae00ac40dcc]
________________________________
From: Aliaksandr (Alex) Zasinets ***@***.***>
Sent: Monday, June 10, 2024 12:55 PM
To: ochaklosh/A59-QA-Automation-Framework ***@***.***>
Cc: Oksana Chaklosh ***@***.***>; Author ***@***.***>
Subject: Re: [ochaklosh/A59-QA-Automation-Framework] Homework19 tests (PR #11)


Hello,

I have checked your homework 19 and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, ***@***.******@***.***>)!

Thumbs-up-hand-with-crown-illustration-on-transparent-background-PNG.png (view on web)<https://github.com/Tootienahs/A59-QA-Automation-Framework/assets/104761482/20861dce-f8e7-4435-9b92-de6436933d0b>

[Quote]<https://camo.githubusercontent.com/a8dfeec8ed326efd6f2fbc7daac7883588119cae338a95a587b6bf4f2fd86c3e/68747470733a2f2f6769746875622d726561646d652d71756f7465732d6261792e76657263656c2e6170702f71756f74653f7468656d653d6461726b266c61796f75743d736f637261746573>

image.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/6cb12f29-a13f-4152-b93e-d1f928c30af7>

HINT

image.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/e21deddb-6c9a-43ed-a7c1-1669a796dc68>

https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/232ddda0-432d-42a2-8aad-d002c5ceac1a

click

Screenshot_1.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/2d0d3a39-ed9f-4bfb-9f9d-ee9e6d9929b9>

—
Reply to this email directly, view it on GitHub<https://github.com/ochaklosh/A59-QA-Automation-Framework/pull/11#issuecomment-2158867744>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BIT3MF5QKHZXAAQUNXY4Q5LZGXLAXAVCNFSM6AAAAABJB6OZCOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNJYHA3DONZUGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Berichicko commented:
> Hello Alex, I am sorry to bother you, but when I run deletePlaylist() I get a Tests Failed. Please look at the screen shot. It can't find selector. However, for you test worked. I am confused. I would appreciate if you could help me with this. Thanks, Oksana [cid:827a6050-b990-45f1-9dec-dae00ac40dcc]
> […](#)
> ________________________________ From: Aliaksandr (Alex) Zasinets ***@***.***> Sent: Monday, June 10, 2024 12:55 PM To: ochaklosh/A59-QA-Automation-Framework ***@***.***> Cc: Oksana Chaklosh ***@***.***>; Author ***@***.***> Subject: Re: [ochaklosh/A59-QA-Automation-Framework] Homework19 tests (PR #11) Hello, I have checked your homework 19 and I wanted to let you know that you're doing great! Keep up the excellent work! Best wishes from the HW team (Alex, ***@***.******@***.***>)! Thumbs-up-hand-with-crown-illustration-on-transparent-background-PNG.png (view on web)<https://github.com/Tootienahs/A59-QA-Automation-Framework/assets/104761482/20861dce-f8e7-4435-9b92-de6436933d0b> [Quote]<https://camo.githubusercontent.com/a8dfeec8ed326efd6f2fbc7daac7883588119cae338a95a587b6bf4f2fd86c3e/68747470733a2f2f6769746875622d726561646d652d71756f7465732d6261792e76657263656c2e6170702f71756f74653f7468656d653d6461726b266c61796f75743d736f637261746573> image.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/6cb12f29-a13f-4152-b93e-d1f928c30af7> HINT image.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/e21deddb-6c9a-43ed-a7c1-1669a796dc68> https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/232ddda0-432d-42a2-8aad-d002c5ceac1a click Screenshot_1.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/2d0d3a39-ed9f-4bfb-9f9d-ee9e6d9929b9> — Reply to this email directly, view it on GitHub<[#11 (comment)](https://github.com/ochaklosh/A59-QA-Automation-Framework/pull/11#issuecomment-2158867744)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BIT3MF5QKHZXAAQUNXY4Q5LZGXLAXAVCNFSM6AAAAABJB6OZCOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNJYHA3DONZUGQ>. You are receiving this because you authored the thread.Message ID: ***@***.***>

Hi dear Oksana

Big delays in tests can make them slow and unreliable. A 10-second delay might not always be needed and could make tests unstable. 

I set it to 2000 ( 2 sec) and the test passed

![image](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/2a74c509-4dc7-43aa-917b-9c3df5b23d2b)


![image](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/5d0500c9-5d60-4637-bf83-dfbdcd485410)

ochaklosh commented:
Thank you so much Alex for your advice.
I initially changed sleep to 2000 like you suggested. However it gave me a different fail due to the create message instead of delete one.
I increased sleep to 10000 after creating a playlist and left the other ones at 2000. It worked. Again, thank you for your time and willingness to help.
Oksana
________________________________
From: Aliaksandr (Alex) Zasinets ***@***.***>
Sent: Monday, June 10, 2024 2:01 PM
To: ochaklosh/A59-QA-Automation-Framework ***@***.***>
Cc: Oksana Chaklosh ***@***.***>; Author ***@***.***>
Subject: Re: [ochaklosh/A59-QA-Automation-Framework] Homework19 tests (PR #11)


Hello Alex, I am sorry to bother you, but when I run deletePlaylist() I get a Tests Failed. Please look at the screen shot. It can't find selector. However, for you test worked. I am confused. I would appreciate if you could help me with this. Thanks, Oksana [cid:827a6050-b990-45f1-9dec-dae00ac40dcc]
…
________________________________ From: Aliaksandr (Alex) Zasinets @.> Sent: Monday, June 10, 2024 12:55 PM To: ochaklosh/A59-QA-Automation-Framework @.> Cc: Oksana Chaklosh @.>; Author @.> Subject: Re: [ochaklosh/A59-QA-Automation-Framework] Homework19 tests (PR #11<https://github.com/ochaklosh/A59-QA-Automation-Framework/pull/11>) Hello, I have checked your homework 19 and I wanted to let you know that you're doing great! Keep up the excellent work! Best wishes from the HW team (Alex, @.@.>)! Thumbs-up-hand-with-crown-illustration-on-transparent-background-PNG.png (view on web)https://github.com/Tootienahs/A59-QA-Automation-Framework/assets/104761482/20861dce-f8e7-4435-9b92-de6436933d0b [Quote]https://camo.githubusercontent.com/a8dfeec8ed326efd6f2fbc7daac7883588119cae338a95a587b6bf4f2fd86c3e/68747470733a2f2f6769746875622d726561646d652d71756f7465732d6261792e76657263656c2e6170702f71756f74653f7468656d653d6461726b266c61796f75743d736f637261746573 image.png (view on web)https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/6cb12f29-a13f-4152-b93e-d1f928c30af7 HINT image.png (view on web)https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/e21deddb-6c9a-43ed-a7c1-1669a796dc68 https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/232ddda0-432d-42a2-8aad-d002c5ceac1a click Screenshot_1.png (view on web)https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/2d0d3a39-ed9f-4bfb-9f9d-ee9e6d9929b9 — Reply to this email directly, view it on GitHub<#11 (comment)<https://github.com/ochaklosh/A59-QA-Automation-Framework/pull/11#issuecomment-2158867744>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/BIT3MF5QKHZXAAQUNXY4Q5LZGXLAXAVCNFSM6AAAAABJB6OZCOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNJYHA3DONZUGQ. You are receiving this because you authored the thread.Message ID: @.***>

Hi dear Oksana

Big delays in tests can make them slow and unreliable. A 10-second delay might not always be needed and could make tests unstable.

I set it to 2000 ( 2 sec) and the test passed

image.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/2a74c509-4dc7-43aa-917b-9c3df5b23d2b>

image.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/5d0500c9-5d60-4637-bf83-dfbdcd485410>

—
Reply to this email directly, view it on GitHub<https://github.com/ochaklosh/A59-QA-Automation-Framework/pull/11#issuecomment-2158976480>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BIT3MF5N3IGHUW32Q2FRDALZGXSXFAVCNFSM6AAAAABJB6OZCOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNJYHE3TMNBYGA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

ochaklosh committed:
Update Homework19Tests.java

Changed Thread.sleep from 10000 to 2000 everywhere except after the playList was created. Before with 10000 all over it was giving the Fail Test, too much time. After I reset everywhere to 2000 I got another Fail due to the success messages being displayed at the same time and the actual value of the message was created playlist instead of deleted. By increasing the sleep time after creation of playlist to 10000 it allowed for the correct success message to be stored in the variable.

ochaklosh commented:
Hello Alex,
One more question. Regretfully, I didn't ask it during Friday class, but I was behind and still lost.
We were discussing waits

public void loginWithValidEmailEmptyPassword() throws InterruptedException {
    //navigateToPage();
    enterEmail(email);
    submit();

    Thread.sleep(2000); //sleep or pause for 2 seconds (adjust as needed)
    //Expected Result
    Assert.assertEquals(driver.getCurrentUrl(),url);
}

In this instance which explicit wait would you suggest for me to use if I want to replace the hard stop by Thread.sleep(2000)?
Should I remove those stops if I know that program works? It is a more like a visual to be able to see the actions on the screen.
Thanks,
OKsana
________________________________
From: Oksana Chaklosh ***@***.***>
Sent: Monday, June 10, 2024 3:01 PM
To: ochaklosh/A59-QA-Automation-Framework ***@***.***>
Subject: Re: [ochaklosh/A59-QA-Automation-Framework] Homework19 tests (PR #11)

Thank you so much Alex for your advice.
I initially changed sleep to 2000 like you suggested. However it gave me a different fail due to the create message instead of delete one.
I increased sleep to 10000 after creating a playlist and left the other ones at 2000. It worked. Again, thank you for your time and willingness to help.
Oksana
________________________________
From: Aliaksandr (Alex) Zasinets ***@***.***>
Sent: Monday, June 10, 2024 2:01 PM
To: ochaklosh/A59-QA-Automation-Framework ***@***.***>
Cc: Oksana Chaklosh ***@***.***>; Author ***@***.***>
Subject: Re: [ochaklosh/A59-QA-Automation-Framework] Homework19 tests (PR #11)


Hello Alex, I am sorry to bother you, but when I run deletePlaylist() I get a Tests Failed. Please look at the screen shot. It can't find selector. However, for you test worked. I am confused. I would appreciate if you could help me with this. Thanks, Oksana [cid:827a6050-b990-45f1-9dec-dae00ac40dcc]
…
________________________________ From: Aliaksandr (Alex) Zasinets @.> Sent: Monday, June 10, 2024 12:55 PM To: ochaklosh/A59-QA-Automation-Framework @.> Cc: Oksana Chaklosh @.>; Author @.> Subject: Re: [ochaklosh/A59-QA-Automation-Framework] Homework19 tests (PR #11<https://github.com/ochaklosh/A59-QA-Automation-Framework/pull/11>) Hello, I have checked your homework 19 and I wanted to let you know that you're doing great! Keep up the excellent work! Best wishes from the HW team (Alex, @.@.>)! Thumbs-up-hand-with-crown-illustration-on-transparent-background-PNG.png (view on web)https://github.com/Tootienahs/A59-QA-Automation-Framework/assets/104761482/20861dce-f8e7-4435-9b92-de6436933d0b [Quote]https://camo.githubusercontent.com/a8dfeec8ed326efd6f2fbc7daac7883588119cae338a95a587b6bf4f2fd86c3e/68747470733a2f2f6769746875622d726561646d652d71756f7465732d6261792e76657263656c2e6170702f71756f74653f7468656d653d6461726b266c61796f75743d736f637261746573 image.png (view on web)https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/6cb12f29-a13f-4152-b93e-d1f928c30af7 HINT image.png (view on web)https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/e21deddb-6c9a-43ed-a7c1-1669a796dc68 https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/232ddda0-432d-42a2-8aad-d002c5ceac1a click Screenshot_1.png (view on web)https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/2d0d3a39-ed9f-4bfb-9f9d-ee9e6d9929b9 — Reply to this email directly, view it on GitHub<#11 (comment)<https://github.com/ochaklosh/A59-QA-Automation-Framework/pull/11#issuecomment-2158867744>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/BIT3MF5QKHZXAAQUNXY4Q5LZGXLAXAVCNFSM6AAAAABJB6OZCOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNJYHA3DONZUGQ. You are receiving this because you authored the thread.Message ID: @.***>

Hi dear Oksana

Big delays in tests can make them slow and unreliable. A 10-second delay might not always be needed and could make tests unstable.

I set it to 2000 ( 2 sec) and the test passed

image.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/2a74c509-4dc7-43aa-917b-9c3df5b23d2b>

image.png (view on web)<https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/5d0500c9-5d60-4637-bf83-dfbdcd485410>

—
Reply to this email directly, view it on GitHub<https://github.com/ochaklosh/A59-QA-Automation-Framework/pull/11#issuecomment-2158976480>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BIT3MF5N3IGHUW32Q2FRDALZGXSXFAVCNFSM6AAAAABJB6OZCOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNJYHE3TMNBYGA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
